### PR TITLE
Update hcp sdk

### PR DESCRIPTION
**Pull request by uruemu**
### :hammer_and_wrench: Description

Update `hcp-go-sdk` to the latest version. Found breaking change with the vault API and applied client side updates to remediate the issues.

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
➜  terraform-provider-hcp git:(update-hcp-sdk) ✗ make testacc TESTARGS='-run=TestAccVaultClusterAWS'
==> Checking that code complies with gofmt requirements...
golangci-lint run --config ./golangci-config.yml 
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultClusterAWS -timeout 360m -parallel=10
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/clients    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/consul     (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/input      (cached) [no tests to run]
=== RUN   TestAccVaultClusterAWS
=== PAUSE TestAccVaultClusterAWS
=== CONT  TestAccVaultClusterAWS
--- PASS: TestAccVaultClusterAWS (3372.04s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   3373.094s
➜  terraform-provider-hcp git:(update-hcp-sdk) ✗ make testacc TESTARGS='-run=TestAccVaultClusterAzure'
==> Checking that code complies with gofmt requirements...
golangci-lint run --config ./golangci-config.yml 
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultClusterAzure -timeout 360m -parallel=10
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/clients    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/consul     (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/input      (cached) [no tests to run]
=== RUN   TestAccVaultClusterAzure
=== PAUSE TestAccVaultClusterAzure
=== CONT  TestAccVaultClusterAzure
--- PASS: TestAccVaultClusterAzure (3470.10s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   3471.011s
➜  terraform-provider-hcp git:(update-hcp-sdk) ✗ 
```

### CODE REVIEW

File: internal/provider/resource_vault_cluster.go [hunk 1]
```diff
@@ -499,7 +528,7 @@ func resourceVaultClusterRead(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	// The Vault cluster was already deleted, remove from state.
-	if *cluster.State == vaultmodels.HashicorpCloudVault20201125ClusterStateDELETED {
+	if *cluster.State == vaultmodels.HashicorpCloudVault20201125ClusterStateDELETING {
```
Comment (codergs) [inline internal/provider/resource_vault_cluster.go:531]: nit: DELETED

File: internal/clients/vault_cluster.go [hunk 2]
```diff
@@ -118,11 +126,19 @@ func UpdateVaultClusterPublicIps(ctx context.Context, client *Client, loc *share
 func UpdateVaultMajorVersionUpgradeConfig(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, clusterID string,
 	config *vaultmodels.HashicorpCloudVault20201125MajorVersionUpgradeConfig) (vaultmodels.HashicorpCloudVault20201125UpdateMajorVersionUpgradeConfigResponse, error) {
 
+	locInternal := &vaultmodels.HashicorpCloudInternalLocationLocation{
+		OrganizationID: loc.OrganizationID,
+		ProjectID:      loc.ProjectID,
+		Region: &vaultmodels.HashicorpCloudInternalLocationRegion{
+			Provider: loc.Region.Provider,
```
Comment (helenfufu) [inline internal/clients/vault_cluster.go:133]: This line 133 errored on our acceptance tests with a nil pointer dreference panic. It appears either `loc` is nil or `loc.Region` is nil, which is why it's unable to get `loc.Region.Provider`

AWS
```zsh
$ make testacc TESTARGS='-run=TestAccVaultClusterAWS'
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultClusterAWS -timeout 360m -parallel=10
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/clients	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAccVaultClusterAWS
=== PAUSE TestAccVaultClusterAWS
=== CONT  TestAccVaultClusterAWS
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105a004fc]

goroutine 2889 [running]:
github.com/hashicorp/terraform-provider-hcp/internal/clients.UpdateVaultMajorVersionUpgradeConfig({0x105ecf988?, 0x1400091de90}, 0x140007ce5b0, 0x1400091df80, {0x14000648d66, 0x1b}, 0x1400066f860)
	/Users/helenfu/workspace/terraform-provider-hcp/internal/clients/vault_cluster.go:133 +0xbc
github.com/hashicorp/terraform-provider-hcp/internal/provider.resourceVaultClusterUpdate({0x105ecf988, 0x1400091de90}, 0x1400046e400, {0x105d20240?, 0x140007ce5b0})
	/Users/helenfu/workspace/terraform-provider-hcp/internal/provider/resource_vault_cluster.go:637 +0xa5c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x14000693500, {0x105ecf9c0, 0x14000509d10}, 0xd?, {0x105d20240, 0x140007ce5b0})
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:741 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000693500, {0x105ecf9c0, 0x14000509d10}, 0x140001f7790, 0x140005b9f80, {0x105d20240, 0x140007ce5b0})
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:847 +0x68c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140008c10f8, {0x105ecf9c0?, 0x14000509c50?}, 0x14000886730)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xb78
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000afbea0, {0x105ecf9c0?, 0x14000509440?}, 0x14000275e30)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x105e72740?, 0x14000afbea0}, {0x105ecf9c0, 0x14000509440}, 0x14000275d50, 0x0)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140007723c0, {0x105ed3c78, 0x140004449c0}, 0x14000247d40, 0x14000242c30, 0x10673efe0, 0x0)
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0xbac
google.golang.org/grpc.(*Server).handleStream(0x140007723c0, {0x105ed3c78, 0x140004449c0}, 0x14000247d40, 0x0)
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x28c
FAIL	github.com/hashicorp/terraform-provider-hcp/internal/provider	2056.307s
FAIL
make: *** [testacc] Error 1
```

Azure
```zsh
$ make testacc TESTARGS='-run=TestAccVaultClusterAzure'
TF_ACC=1 go test ./internal/... -v -run=TestAccVaultClusterAzure -timeout 360m -parallel=10
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/clients	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/consul	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/input	(cached) [no tests to run]
=== RUN   TestAccVaultClusterAzure
=== PAUSE TestAccVaultClusterAzure
=== CONT  TestAccVaultClusterAzure
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1032c04fc]

goroutine 2765 [running]:
github.com/hashicorp/terraform-provider-hcp/internal/clients.UpdateVaultMajorVersionUpgradeConfig({0x10378f988?, 0x14000458c90}, 0x140007fc4e0, 0x14000458db0, {0x1400066c126, 0x1d}, 0x14000705470)
	/Users/helenfu/workspace/terraform-provider-hcp/internal/clients/vault_cluster.go:133 +0xbc
github.com/hashicorp/terraform-provider-hcp/internal/provider.resourceVaultClusterUpdate({0x10378f988, 0x14000458c90}, 0x140009ea580, {0x1035e0240?, 0x140007fc4e0})
	/Users/helenfu/workspace/terraform-provider-hcp/internal/provider/resource_vault_cluster.go:637 +0xa5c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x14000567420, {0x10378f9c0, 0x140003d4ab0}, 0xd?, {0x1035e0240, 0x140007fc4e0})
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:741 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000567420, {0x10378f9c0, 0x140003d4ab0}, 0x1400087e680, 0x140009ea400, {0x1035e0240, 0x140007fc4e0})
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:847 +0x68c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000b6ea50, {0x10378f9c0?, 0x140003d49f0?}, 0x1400059e1e0)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xb78
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000565c20, {0x10378f9c0?, 0x140003d41e0?}, 0x140001a4230)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x103732740?, 0x14000565c20}, {0x10378f9c0, 0x140003d41e0}, 0x140001a4000, 0x0)
	/Users/helenfu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400074e000, {0x103793c78, 0x140008001a0}, 0x14000660b40, 0x14000c21c20, 0x103ffefe0, 0x0)
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0xbac
google.golang.org/grpc.(*Server).handleStream(0x1400074e000, {0x103793c78, 0x140008001a0}, 0x14000660b40, 0x0)
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/helenfu/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x28c
FAIL	github.com/hashicorp/terraform-provider-hcp/internal/provider	2558.218s
FAIL
make: *** [testacc] Error 1
```

Comment (helenfufu) [reply internal/clients/vault_cluster.go:133]: Believe this is the issue: The `loc` we are passing into `UpdateVaultMajorVersionUpgradeConfig` from [here](https://github.com/hashicorp/terraform-provider-hcp/blob/615473ecf3aac95c53421dbfac775b33be09ad17/internal/provider/resource_vault_cluster.go#L307-L310) does not have `Region` populated.

File: internal/provider/resource_vault_cluster.go [hunk 3]
```diff
@@ -823,7 +869,21 @@ func setVaultClusterResourceData(d *schema.ResourceData, cluster *vaultmodels.Ha
 	if cluster.PerformanceReplicationInfo != nil {
 		prInfo := cluster.PerformanceReplicationInfo
 		if prInfo.PrimaryClusterLink != nil {
-			primaryLink, err := linkURL(cluster.PerformanceReplicationInfo.PrimaryClusterLink)
+			primaryClusterLink := &sharedmodels.HashicorpCloudLocationLink{
```
Comment (codergs) [inline internal/provider/resource_vault_cluster.go:872]: can we consider moving this translation to another func? that will allow us to re-use this in other places.

File: internal/clients/vault_cluster.go [hunk 4]
```diff
@@ -118,11 +126,23 @@ func UpdateVaultClusterPublicIps(ctx context.Context, client *Client, loc *share
 func UpdateVaultMajorVersionUpgradeConfig(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, clusterID string,
 	config *vaultmodels.HashicorpCloudVault20201125MajorVersionUpgradeConfig) (vaultmodels.HashicorpCloudVault20201125UpdateMajorVersionUpgradeConfigResponse, error) {
 
+	region := &sharedmodels.HashicorpCloudLocationRegion{}
+	if loc.Region != nil {
+		region = loc.Region
+	}
+	locInternal := &vaultmodels.HashicorpCloudInternalLocationLocation{
```
Comment (codergs) [inline internal/clients/vault_cluster.go:133]: nit: same for this - move to a small func

File: internal/provider/resource_vault_cluster.go (hunk 1 above)
Comment (codergs) [reply internal/provider/resource_vault_cluster.go:531]: I think we should remove this check as DELETE can never happen :( somehow proto field was removed here.

**Review by codergs — APPROVED** [review]
LGTM. Small nits around creating small wrapper func to reduce copy-pasta or duplication.

**Pull request merged**